.github/workflows/ci.yml

```diff
@@ -101,11 +109,12 @@ jobs:
           CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
           ASYNC_STD_THREAD_COUNT: 4
 
-      - name: Run tests shared-memory
+      - name: Run tests with SHM
+        if: ${{ matrix.os == 'ubuntu-latest' }}
```

Comment:
I'm not fond of having filtering at CI level. We can better tune it in the code via conditional compilation.

Reply:
The main reason why I used CI filtering is that transport_shm feature is not allowed for windows. But I think it would be easy to add windows support for transport_shm feature in the future (using windows named pipes that really seems to be an easy task).

Reply:
Shouldn't be `if: ${{ matrix.os == 'ubuntu-latest' || matrix.os == 'macOS-latest' }}`? Including also `macOS-latest` as target?